### PR TITLE
Simpler examples of concurrency primitives

### PR DESCRIPTION
**Pull request by LLCampos**
~~Didn't add an example for Queue because [it is being moved to cats-effect](https://github.com/typelevel/fs2/issues/2213). :)~~

### CODE REVIEW

File: site/concurrency-primitives.md [hunk 1]
```diff
@@ -6,7 +6,61 @@ In the [`fs2.concurrent` package](https://github.com/functional-streams-for-scal
...
-These data structures could be very handy in a few cases. See below examples of each of them originally taken from [gvolpe examples](https://github.com/gvolpe/advanced-http4s/tree/master/src/main/scala/com/github/gvolpe/fs2):
+## Simple Examples
+
+### Topic
+
+(based on [Pera Villega](https://perevillega.com/)'s example [here](https://underscore.io/blog/posts/2018/03/20/fs2.html))
+
+Topic implements the publish-subscribe pattern. In the following example, `publisher` and `subscriber` are started concurrently with the publisher continuously publishing the string `"1"` to the topic and the subscriber consuming it until it received four elements, including the initial element, the string `"Topic start"`.
+
+```scala
```
Comment (mpilquist) [inline site/concurrency-primitives.md:17]: If you change this to `scala mdoc`, the example will get compiled/tested as part of site generation. In this case, it'll show that the example is using the old CE2 / FS2 2.x API and hence needs to be updated for CE3 / FS2 3.x.

File: site/concurrency-primitives.md [hunk 2]
```diff
@@ -6,7 +6,61 @@ In the [`fs2.concurrent` package](https://github.com/functional-streams-for-scal
...
+}.unsafeRunSync()
+```
+
+### Signal
+
+`Signal` can be used as a means of communication between two different streams. In the following example `s1` is a stream that emits the current time every second and is interrupted when the signal has value of `true`. `s2` is a stream that sleeps for four seconds before assigning `true` to the signal. This leads to `s1` being interrupted.
+
+(Note that `SignallingRef` extends `Signal`)
+
+```scala
```
Comment (mpilquist) [inline site/concurrency-primitives.md:41]: Same as above, needs updates for CE3

**Review by mpilquist — CHANGES_REQUESTED** [review]
Thanks for working on docs! Added a couple comments about how we're targeting CE3/FS2 3.x now.

We could add docs on `Queue` too, given that FS2 has some methods for working with `cats.effect.std.Queue` -- e.g., `Stream.fromQueueNoneTerminated` being the most common.

**Review by diesalbla — APPROVED** [review]
Thanks!

**Comment by LLCampos** [comment]
@mpilquist, adapted things for CE3/FS2 3.x. Should I also change the branch this is being merge into? :thinking:

**Comment by LLCampos** [comment]
Hm... CI workflow failed... is this expected?

File: site/concurrency-primitives.md [hunk 3]
```diff
@@ -14,18 +15,15 @@ In the [`fs2.concurrent` package](https://github.com/functional-streams-for-scal
...
+import cats.effect.unsafe.implicits.global
 import fs2.Stream
 import fs2.concurrent.Topic
 
-import scala.concurrent.ExecutionContextExecutor
-
-implicit val ec: ExecutionContextExecutor = scala.concurrent.ExecutionContext.global
-implicit val cs: ContextShift[IO] = IO.contextShift(ec)
-implicit val c: ConcurrentEffect[IO] = IO.ioConcurrentEffect
+val topic: IO[Topic[IO, String]] = Topic("Topic start")(Concurrent[IO])
```
Comment (ritschwumm) [inline site/concurrency-primitives.md:24]: imho typeclass instances should never be passed around manually, except as a workaround where more than one instance exists. especially not in example code that will be widely copied.
if it works, i'd suggesst writing `Topic[IO,String](...)` instead.

**Pull request merged**